### PR TITLE
[Enterprise Billing] Refactor / reuse subscription validation lib

### PR DESCRIPTION
**Pull request by philipperolet**
## Description
- Switches validity checks from @flvndvd to the validation lib
- validation lib now uses the global constant defined by @flvndvd in types
- extra: checks the subscription is unassigned in poke enterprise upgrade logic, ping @PopDaph wdyt?

### CODE REVIEW

File: front/lib/plans/stripe.ts [hunk 1]
```diff
@@ -249,64 +253,34 @@ export async function cancelSubscriptionImmediately({
   return true;
 }
 
-const REPORT_USAGE_VALUES = ["MAU_1", "MAU_5", "MAU_10", "PER_SEAT"];
-
 /**
  * Checks that a subscription created in Stripe is usable by Dust, returns an
  * error otherwise.
  */
 export function assertStripeSubscriptionValid(
```
Comment (flvndvd) [inline front/lib/plans/stripe.ts:260]: `nit`
```suggestion
export function assertStripeSubscriptionIsValid(
```

File: front/lib/plans/stripe.ts [hunk 2]
```diff
@@ -249,64 +253,34 @@ export async function cancelSubscriptionImmediately({
...
-        item.price.recurring.interval !== "month" ||
-        item.price.recurring.interval_count !== 1
-      ) {
-        return new Err({
-          invalidity_message:
-            "Subscription recurring price has invalid interval, only 1-month intervals are allowed.",
-        });
-      }
+  for (const item of itemsToCheck) {
+    const itemValidation = assertStripeSubscriptionItemValid({ item });
```
Comment (flvndvd) [inline front/lib/plans/stripe.ts:281]:
```suggestion
    const itemValidation = assertStripeSubscriptionItemIsValid({ item });
```

**Review by flvndvd — APPROVED** [review]
LGTM, left 2 nits.

File: front/pages/api/poke/workspaces/[wId]/upgrade_enterprise.ts [hunk 3]
```diff
@@ -77,6 +80,21 @@ async function handler(
           },
         });
       }
+
+      const subscription = await getSubscriptionForStripeId(
```
[the remark this answers is not in the document]
Comment (PopDaph) [reply front/pages/api/poke/workspaces/[wId]/upgrade_enterprise.ts:84]: LGTM

**Pull request merged**